### DISCUSSION TITLE
Suggestions on Education VC specs

### DISCUSSION THREAD

**Discussion by AnuragSingh55** [INCOMS/vc-specs#10] [Ideas & Enhacnements]
Hey everyone, wanted to use this forum to share with you all VC Specs which we are working on for education credentials, as part of one of our programs. This includes specs for 3 types of credentials, namely;

1. ProofOfEnrollment – A credential that seeks to define which education institution a particular student is linked to (school ID card as a credential)
2. ProofOfAssessment – This seeks to provide assessment records to a student (report card as a credential)
3. ProofOfBenefits – This credential provides a list of benefits/ aid/scholarships that a student avails (benefit/ scholarship as a credential)

![ULP-Page-27](https://user-images.githubusercontent.com/106961941/217565108-fdbda62f-4109-4466-8cd2-7af2b4104f3c.jpg)

Would request the community to provide feedback on these by 10th Feb, so that we can include suggestions and concerns and ensure our implementation meets its timelines. Thanks again.

@parthlawate @coolbung @girirajdaga

cc - @suresh12 @ChakshuGautam  @surendrasinghs

**Comment by sankarshanmukhopadhyay** [comment]
How would accomplishments (viz. participant/winner in a competition, representative at an event etc) fit in?

**Reply by AnuragSingh55** [reply]
Thanks for the input, will plan to add another credential type (ProofOfAchievements perhaps?) which can be like an achievement badge for students (can be given for academic/ non-academic accomplishment and recognition for high attendance/ responsibility etc.)

**Comment by surendrasinghs** [comment]
Considering these specs are not just for formal school education, rather for the larger landscape of learning, schooling, training, skilling and livelihoods use-cases, please look at the following suggestions:

1. Institute Name instead of School Name for Proof of Enrolment & Assessment
2. Date / Month / Year of Examination instead of Academic Year. Same for "term" under Examination.
3. Does benefit always need "amount"? What if it's a direct delivery of goods and not cash?
4. Are we assuming the issuer to be the assessor (for Proof of Assessment) or should the assessor be captured & represented distinctly from the issuer? Same for benefits - Benefits sponsored by vs Benefits delivered by (e.g. Tata Trust scholarship delivered through State education system)? => I think, in the spirit of minimalism, we should just stick to issuer - issuers can express much of this 'transactional' detail if they choose to - but the spec should not assume/mandate it.

**Reply by sankarshanmukhopadhyay** [reply]
For (4) the assessor should be represented as a separate, unique actor. Consider this example - a state transport company can seek a student ID, verify and issue a monthly pass - in this transaction the same actor (state transport company) switches between two distinct roles in order to complete the two different actions.

**Comment by pramodkvarma** [comment]
Few points. Following are the credentials that we should "generalize" (not specific to schools etc.)

1. Proof of Association (enrollment/membership into any in system - e.g. employee card, student card, sports/cultural/other club memberships, benefit card, etc)
2. Proof of Skill/Learning
3. Proof of Work (any type of work/project/etc)
4. Proof of Achievement (recognition of any type of work/achievement/scholarships/etc)
5. Proof of Earning
6. Proof of License (or entitlement or authorization - e.g. Entry certificate, any type of licenses, etc.)

With the above in mind, I would not have ProofOfEnrollment into a school separately, rather generalize into ProofOfAssociation and have attributes that are named in a generalized way. Enrollment into School is one instance of this.

Can we attempt this so that we don't end up producing many schemas that almost look like? Of course, we can create JSON schemas to compose specialized ones from generic JSON schema objects such as "Person", "Location", etc. Please check out [some core objects in beckn protocol](https://github.com/beckn/protocol-specifications/tree/master/schema) and see if we can leverage those.

Can we attempt the above please? @surendrasinghs @tejash-jl

**Comment by surendrasinghs** [comment]
Thanks, @pramodkvarma and others for comments. Here's a sheet to capture everyone's comments on [VC Specs](https://docs.google.com/spreadsheets/d/1y4z1X7Dfercj7C3wkKbPAR_ExHbL_KgXbwtFzeFK078/edit#gid=0)

We have taken a shot at drafting Proof of Association to start with. Please suggest changes to the specification as well as the methodology/template for capturing comments.

We aim to get to Proof of Assessment (e.g. Exam Report Card, Regular Assessments) and Proof of Payment (e.g. Benefit Transfer, Salary slips) soon after, since we are looking at some of these use-cases in the immediate future.

Please add any other category/type of Proofs in the "use-cases" sheet. Look forward to contributions, comments, and suggestions. 

Here's the relevant thread where larger community and various use-cases are being discussed: [vc-specs/discussions/11](https://github.com/VC-Specs/vc-specs/discussions/11)

Thanks!

**Comment by AnuragSingh55** [comment]
Hey everyone, we were able to close a few more [specs](https://docs.google.com/spreadsheets/d/1y4z1X7Dfercj7C3wkKbPAR_ExHbL_KgXbwtFzeFK078/edit#gid=732344055) including proof of association, proof of benefits, proof of work, proof of achievement. On this thread. Would be great to receive more comments on this front. Thanks!

**Reply by pramodkvarma** [reply]
I would like a review of latest specs when I am back in India. Can @surendrasinghs facilitate that please? Thanks.

**Comment by raj7desai** [comment]
Hello everyone! 

One of the widespread issues that's happening in up-skilling ecosystem is that every institute claims to build various kinds of skills and issues certificates or credentials for the same. However, the authenticity of these certificates often remains questionable.

I would suggest that we should definitely have a Proof of Skill spec which is verified by independent evaluators to ensure the assessments are unbiased and can be used by various organizations to evaluate and hire candidates. Think of this as a new form of a CV or a resume.

A decentralized  credentialing mechanism would really help ensure that institutes will back their claims with legit skill development. The other advantage of this is that hiring becomes extremely easy and seamless as companies will have an assessment of skill they can trust. 

We have created a design of the Proof of Skill protocol based on Soul Bound Tokens and anonymous consensus amongst Skill Validators. I am exploring how can we integrate these with various others VCs we have developed so far. Any help/comments or guidance on this will be highly appreciated. Thanks
